cranelift/interpreter/src/value.rs

```diff
@@ -365,6 +380,14 @@ impl Value for DataValue {
         binary_match!(%(&self, &other); [I8, I16, I32, I64])
     }
 
+    fn add_sat(self, other: Self) -> ValueResult<Self> {
+        binary_match!(saturating_add(self, &other); [I8, I16, I32, I64, I128, U8, U16, U32, U64, U128])
+    }
+
+    fn sub_sat(self, other: Self) -> ValueResult<Self> {
+        binary_match!(saturating_sub(self, &other); [I8, I16, I32, I64, I128, U8, U16, U32, U64, U128])
+    }
```

Comment:
What do you think about implementing these instructions as: convert to larger type, use already existing add/sub, convert back to the right type with saturation (maybe with a new saturating `ValueConversionKind`). I can't remember how many instructions end up being saturating ones but if there are more it could reduce the size of `DataValue`.

Reply:
Looking at the list of opcodes, there's a few but given that a lot of them are signed/unsigned variants of each other (which I don't think will need re-implementing each time, as in this case for example), I think there'd be ~5 saturating ops in DataValue. We could do that suggestion though if preferred, which might make implementing opcodes like the `narrow` set easier later (the only question is if `[I|U]128` is supposed to be supported; I see a few of the other functions in DataValue seem to ignore it and being the typically largest integer type would require different implementation).

Reply:
Yeah, it sounds like the current approach will be fine.

Reply:
Thanks for adding these instructions!